PD-1055 / 25.04 / Pd 1055 add fast deduplication content (#3438)
truenas/documentation: content/SCALE/SCALETutorials/Datasets/DatasetsSCALE.md

### Setting Datasets Access Controls
For information on ACL settings see [Setting Up Permissions]({{< relref "PermissionsSCALE.md" >}}).

### Adding Deduplication
Deduplication is found on the **Add Datasets Advanced Settings** screen.

{{< hint type=info >}}
Best practice is to add deduplication when you first create the dataset.

You can add deduplication to an existing dataset but existing files do not have deduplication applied to them.

> Adding deduplication to an existing dataset only applies deduplication to duplicate files added after you enable the function.

Reviewer: ```suggestion
Adding deduplication to an existing dataset only applies deduplication to blocks written after you enable the function.
```

Reviewer: I have no objections, even though IMHO a bit over-cautious.

> When adding deduplication to a dataset that does not have a deduplication VDEV added to the pool, the deduplication table (DDT) is stored in the dataset with the feature enabled. This decreases the amount of available storage in the dataset.
> To store the DDT in the dedup VDEV and outside of the dataset, add a deduplicaiton VDEV to the pool.

Reviewer: ```suggestion
When enabling deduplication for a dataset of a pool that does not have a deduplication or special VDEV, the deduplication table (DDT) is stored on a regular VDEVs of the pool.
To store the DDT outside of the regular VDEVs, add a deduplicaiton or special VDEV to the pool.
```

Reviewer: Fixed, even though not mine. ;)

{{< /hint >}}
To add deduplication to a new dataset, after entering the name and selecting the dataset preset, click **Advanced Settings**.

To add deduplication to an existing dataset, select the dataset on the **Dataset** screen tree table, click **Edit** on the **Dataset Details** widget to open the **Edit Dataset** screen. Click **Advanced Settings**.

> Scroll down to the **ZFS Deduplication** setting, then change **Inherit (ON)** to **On**.

Reviewer 2: **Inherit (ON)** means it was already enabled in the parent dataset

Author: I've added him as a reviewer per your suggestion. Thanks!!

> Scroll down to the **ZFS Deduplication** setting, then change **Inherit (ON)** to **On**. **Inherit (ON)** means it is already enabled in the parent dataset

Reviewer 2: @micjohnson777 

Perhaps:
Scroll down to the **ZFS Deduplication** setting, then change to **On**.

Perhaps that will suffice? The previous value does not matter and depends on configuration.

Author: next commit makes the change.
